paws_collector.js

```diff
@@ -263,6 +266,93 @@ class PawsCollector extends AlAwsCollector {
             return super.handleEvent(event);
         }
     };
+
+    // check the status of the state in DDB to avoid duplication
+    // there is a lot of logic here. not sur eif there is a better way of deduping. trying for an MVP
+    checkStateSqsMessage(stateSqsMsg, asyncCallback) {
+        const collector = this;
+        const DDB = new AWS.DynamoDB();
+        const tableName = process.env.pawsDDBTableName;
+
+        const params = {
+            Key: {
+                "CollectorId": {S: collector._collectorId},
+                "MessageId": {S: stateSqsMsg.md5OfBody}
+            },
+            TableName: tableName,
+            ConsistentRead: true
+        }
+
+        const getItemPromise = DDB.getItem(params).promise();
+
+        getItemPromise.then(data => {
+            // if the item is alread there, try and see if it is a duplicate
+            if (data.Item) {
+                const Item = data.Item;
+                if(Item.Status.S === STATE_RECORD_INCOMPLETE && moment().unix() - parseInt(Item.Updated.N) < 900) {
+                    console.log(`Duplicate state: ${Item.MessageId.S}, already in progress. skipping`);
+                    return asyncCallback('State is currently being processed by another invocation');
+                } else if (Item.Status.S === STATE_RECORD_COMPLETE){
+                    console.log(`Duplicate state: ${Item.MessageId.S}, already processed. skipping`);
+                    return collector.done();
+                } else {
+                    return collector.updateStateDBEntry(stateSqsMsg, STATE_RECORD_INCOMPLETE, asyncCallback);
+                }
+            // otherwise, put a new item in ddb.
+            } else {
+                const newRecord = {
+                    Item: {
+                        CollectorId: {S: collector._collectorId},
+                        MessageId: {S: stateSqsMsg.md5OfBody},
+                        Updated: {N: moment().unix().toString()},
+                        Status: {S: STATE_RECORD_INCOMPLETE},
+                        // setting DDB time to life. This is the same as the sqs queue message retention
+                        ExpireDate: {N: moment().add(14, 'days').unix().toString()}
+                    },
+                    TableName: tableName
+                }
+                DDB.putItem(newRecord, (err) => {
+                    if(err){
+                        return asyncCallback(err);
+                    } else {
+                        return asyncCallback(null)
+                    }
+                });
+            }
+            
+        }).catch(asyncCallback)
+    }
+
+    updateStateDBEntry(stateSqsMsg, Status, asyncCallback) {
+        const collector = this;
+        const DDB = new AWS.DynamoDB();
+        const tableName = process.env.pawsDDBTableName;
```

Comment:
please add DDB table name into Paws [class properties](https://github.com/alertlogic/paws-collector/pull/147/files#diff-0334275db1035801ee0894723546b59aR111) and use it here